app/decorators/decidim/invite_user_form_decorator.rb

```diff
@@ -7,6 +7,21 @@
 Decidim::InviteUserForm.class_eval do
   attribute :area_id, Integer
 
+  alias_method :original_roles_method, :available_roles_for_select
+
+  def available_roles_for_select
+    if current_user.department_admin?
+      Decidim::User::Roles.all.select { |n| n == "department_admin" }.map do |role|
+        [
+          I18n.t("models.user.fields.roles.#{role}", scope: "decidim.admin"),
+          role,
+        ]
+      end
+    else
+      original_roles_method
+    end
+  end
+
   # called from the view
   def available_areas_for_select
```

Comment:
Only current users's area should be assignable, we should also check this in the form or command

Reply:
Check that please

Reply:
But the area is not shown if you are department admin, is it necessary?

![Captura de pantalla de 2021-04-27 08-46-50](https://user-images.githubusercontent.com/75725233/116197570-34732a00-a735-11eb-9e6e-82a43cc3f9bd.png)